src/components/navbar/Navbar/ProfileModal.tsx

```diff
@@ -223,14 +232,20 @@ function AccountContent({
 }
 
 function PrivateKeyContent() {
-  const encodedSecretKey = useMyAccount((state) => state.encodedSecretKey)
+  const { encodedSecretKey, getPrivateKey } = useMyAccount()
   const secretKey = decodeSecretKey(encodedSecretKey ?? '')
 
   const sendEvent = useSendEvent()
   const onCopyClick = () => {
     sendEvent('click copy_private_key_button')
   }
 
+  useEffect(() => {
+    if (!encodedSecretKey) {
+      getPrivateKey()
+    }
+  }, [encodedSecretKey, getPrivateKey])
```

Comment:
this getPrivateKey is kind of like initializer if I understand correctly? If so, I think its not good putting this here.
I think it will be much better if you can just get the private key when you login and put it in state, so it doesn't need anymore initializing process like this call.

Reply:
This is truly not necessary anymore because the store already passes the private key on login